source/macos/AdaptiveCards/AdaptiveCards/Base/BaseProtocol.swift

```diff
@@ -28,3 +28,31 @@ protocol InputHandlingViewProtocol: NSView {
     var key: String { get }
     var isValid: Bool { get }
 }
+
+protocol SelectActionHandlingProtocol {
+    func setupSelectAction(selectAction: ACSBaseActionElement?, rootView: NSView) -> TargetHandler?
```

Comment:
```suggestion
protocol SelectActionHandlingProtocol {
    var target: TargetHandler? { get set }
    func setupSelectAction(_ selectAction: ACSBaseActionElement?, rootView: NSView)
```
I believe all the Views conforming to this protocol must hold the `target` reference.
Also, the second method need not return a target.